test/unit/compute_resource_test.rb

```diff
@@ -170,4 +170,43 @@ def setup
     cr = FactoryGirl.build(:compute_resource)
     refute as_admin { cr.send(:associate_by, 'mac', '11:22:33:44:55:1a') }.readonly?
   end
+
+  test "vm_compute_attributes_for works" do
+    fake_vm = OpenStruct.new
+    fake_vm.attributes = {:cpus => 2}
+    ComputeResource.any_instance.expects(:find_vm_by_uuid).returns(fake_vm)
+    cr = FactoryGirl.build(:compute_resource)
+    returned_vm = cr.vm_compute_attributes_for('test')
+    assert returned_vm[:cpus] == 2
+  end
+
+  test "vm_compute_attributes_for handles interfaces" do
+    fake_vm = OpenStruct.new
+    fake_vm.attributes = {:cpus => 2}
+    fake_vm.interfaces = [
+      OpenStruct.new(:attributes => {:mac => "00:50:56:a5:1f:60"})
+    ]
+    ComputeResource.any_instance.expects(:find_vm_by_uuid).returns(fake_vm)
+    cr = FactoryGirl.build(:compute_resource)
+    returned_vm = cr.vm_compute_attributes_for('test')
+    assert returned_vm[:cpus] == 2
+    refute returned_vm.has_key? :volumes
+    assert returned_vm[:interfaces].length == 1
+    assert returned_vm[:interfaces][0][:mac] == "00:50:56:a5:1f:60"
```

Comment:
You could probably replace all of this with:

    assert_equal {:cpus => 2, :interfaces => [{:mac => '00:55:56:a5:1f:60'}]}, returned_vm